### DISCUSSION TITLE
Using variables in UQL query

### DISCUSSION THREAD

**Discussion by kausar007** [grafana/grafana-infinity-datasource#311] [Q&A] [answered]
Hi, 

I have CSV data that I can parse and display in a table on a panel. I can filter out values based on a date. For example show only the data where field `Start date` is "11/05/2022". 

I can do this using the following query
```
parse-csv | jsonata "$[`Start date`='11/05/2022']"
```
However I wanted to ask if there is way to dynamically get this date value from the dashboard. I know I can retrieve date from Grafana global variable using format `${__from:date:DD/MM/YYYY}`. Can this be used in the above query? If yes then how would I use it?

Thanks

**Comment by yesoreyeram** [comment, marked as answer]
You can use those variables in UQL query. What error you are getting?

**Reply by kausar007** [reply]
I reliased that the global variable needs double quotes. Wasn't sure which part of the query is interpolated first. So tried these two :
```
parse-csv | jsonata "$[`Start date`=${__from:date:DD/MM/YYYY}]"
parse-csv | jsonata '$[`Start date`="${__from:date:DD/MM/YYYY}"]'
```
and got this error "error applying uql query" for both of them

Finally got it working with escaping the double quotes as follows.
```
parse-csv | jsonata "$[`Start date`=\"${__from:date:DD/MM/YYYY}\"]"
```

**Reply by yesoreyeram** [reply]
Glad you got it working

**Reply by yesoreyeram** [reply]
BTW, I can see that your "Start date" field is of string type so that you needed to covert to string. You may need to convert that field into timestamp field before comparison in most cases.

**Reply by kausar007** [reply]
Thanks. For the data I have it doesn't really matter as I just need to do string comparison anyway. It returns the data as long as the date string in a specific format exists. I am not doing any other comparisons.
